components/org.wso2.identity.event.websubhub.publisher/src/main/java/org/wso2/identity/event/websubhub/publisher/internal/ClientManager.java

```diff
@@ -115,43 +124,49 @@ private RequestConfig createRequestConfig() {
-        LOG.debug("PoolingNHttpClientConnectionManager created with maxConnections: " + maxConnections +
-                " and maxConnectionsPerRoute: " + maxConnectionsPerRoute);
-        return poolingHttpClientConnectionMgr;
-    }
-
-    private void addSslContext(HttpAsyncClientBuilder builder) throws WebSubAdapterException {
+    private SSLContext createSSLContext() throws WebSubAdapterException {
 
         try {
-            SSLContext sslContext = SSLContexts.custom()
-            //default trust strategy is used (trusting all certificates in the provided trust store).
+            return SSLContexts.custom()
                     .loadTrustMaterial(WebSubHubAdapterDataHolder.getInstance().getTrustStore(), null)
                     .build();
-            builder.setSSLContext(sslContext);
-            builder.setSSLHostnameVerifier(new DefaultHostnameVerifier());
-            LOG.debug("SSL context and hostname verifier added");
         } catch (NoSuchAlgorithmException | KeyStoreException | KeyManagementException e) {
-            throw WebSubHubAdapterUtil.handleServerException
-                    (WebSubHubAdapterConstants.ErrorMessages.ERROR_CREATING_SSL_CONTEXT, e);
+            throw WebSubHubAdapterUtil.handleServerException(
+                    WebSubHubAdapterConstants.ErrorMessages.ERROR_CREATING_SSL_CONTEXT, e);
+        }
+    }
+
+    private <T> T createPoolingConnectionManager(Class<T> managerType) throws IOException {
+
+        int maxConnections =
+                WebSubHubAdapterDataHolder.getInstance().getAdapterConfiguration().getDefaultMaxConnections();
+        int maxConnectionsPerRoute =
+                WebSubHubAdapterDataHolder.getInstance().getAdapterConfiguration().getDefaultMaxConnectionsPerRoute();
+
+        if (managerType.equals(PoolingNHttpClientConnectionManager.class)) {
+            ConnectingIOReactor ioReactor = new DefaultConnectingIOReactor();
+            PoolingNHttpClientConnectionManager manager = new PoolingNHttpClientConnectionManager(ioReactor);
+            manager.setMaxTotal(maxConnections);
+            manager.setDefaultMaxPerRoute(maxConnectionsPerRoute);
+            LOG.debug("PoolingNHttpClientConnectionManager created with maxConnections: " + maxConnections +
+                    " and maxConnectionsPerRoute: " + maxConnectionsPerRoute);
+            return managerType.cast(manager);
+        } else if (managerType.equals(PoolingHttpClientConnectionManager.class)) {
+            PoolingHttpClientConnectionManager manager = new PoolingHttpClientConnectionManager();
+            manager.setMaxTotal(maxConnections);
+            manager.setDefaultMaxPerRoute(maxConnectionsPerRoute);
+            LOG.debug("PoolingHttpClientConnectionManager created with maxConnections: " + maxConnections +
+                    " and maxConnectionsPerRoute: " + maxConnectionsPerRoute);
+            return managerType.cast(manager);
+        } else {
```

Comment:
This is a private method overall. So this is used within this class only. In that case, do we have a requirement to support both of these connection managers. Why can't we switch to one

Reply:
yes, since for publishing we use async client and for subscription we use collsable HTTP client